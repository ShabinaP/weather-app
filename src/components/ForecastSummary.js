import React from "react";
import PropTypes from "prop-types";

const ForecastSummary = (props) => {
  const { date, temperature, description, icon } = props;

  return (
    <>
      <div className="date"> {date} </div>
      <div className="temperature"> {temperature} </div>
      <div className="description">{description} </div>
      <div className="icon">{icon} </div>
    </>
  );
};

ForecastSummary.propTypes = {
  date: PropTypes.number.isRequired,
  temperature: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.number.isRequired,
};

export default ForecastSummary;
