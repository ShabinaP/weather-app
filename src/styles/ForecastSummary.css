.moreDetails__button {
    display: flex;
    flex-direction: row;
    padding: 12px 29px;
    margin: 0.5rem;
    background-color: white;
    color: black;
    border: 2px solid #002E45;
    border-radius: 10px;
    text-align: center;
    font-size: 15px;

}

.moreDetails__button:hover {
    border: 2px solid #2B7994;
    color: #2B7994;
    cursor: pointer;
    
}

.forecast-summary__icon {
    width: 100px;
   
}

.forecast-summary__date {
    font-weight: bold;
    font-size: 25px;
    
}

.forecast-summary__temperature {
     font-weight: bold;
}