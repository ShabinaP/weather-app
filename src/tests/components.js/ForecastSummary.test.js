import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 3333444,
    description: "Stub description",
    icon: "stubIcon",
    temperature: {
      min: 20,
      max: 30,
    },
}
  it("renders details of the forecast including: date, description, icon and temperature", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
