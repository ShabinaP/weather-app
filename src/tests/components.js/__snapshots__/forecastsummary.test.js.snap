// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders details of the forecast including: date, description, icon and temperature 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary__date"
  >
     3333444 
  </div>
  <div
    class="forecast-summary__temperature"
  >
    30â„ƒ
  </div>
  <div
    class="forecast-summary__description"
  >
    Stub description 
  </div>
  <div
    class="forecast-summary__icon"
  >
    stubIcon 
  </div>
</DocumentFragment>
`;
